### PR TITLE
docs: add weeklyreport 11.30

### PR DESCRIPTION
**Pull request by lijing-21**
add weeklyreport 11.30

### CODE REVIEW

File: website/blog/2021/12/02/weekly-report-1130.md [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+```Nginx
+subs_filter_types text/html text/css text/xml;
+subs_filter st(\d*).example.com $1.example.com ir;
+subs_filter a.example.com s.example.com;
+subs_filter http://$host https://$host;
+```
+
+However, it seems that APISIX's response-rewrite plugin only supports full replacement, which is equivalent to directly replacing all responses with the return set by the plugin, and does not support partial content replacement:
+
+```Bash
```
Comment (yzeng25) [inline website/blog/2021/12/02/weekly-report-1130.md]:
```suggestion
```shell
```

File: website/i18n/zh/docusaurus-plugin-content-blog/2021/12/02/weekly-report-1130.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+```Nginx
+subs_filter_types text/html text/css text/xml;
+subs_filter st(\d*).example.com $1.example.com ir;
+subs_filter a.example.com s.example.com;
+subs_filter http://$host https://$host;
+```
+
+但是 APISIX 的 response-rewrite 插件文档，好像只能支持完整替换，相当于直接将所有响应都换成插件设置的返回，并不支持部分内容替换:
+
+```Bash
```
Comment (yzeng25) [inline website/i18n/zh/docusaurus-plugin-content-blog/2021/12/02/weekly-report-1130.md]:
```suggestion
```shell
```

File: website/i18n/zh/docusaurus-plugin-content-blog/2021/12/02/weekly-report-1130.md [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+有个支持正则替换的库（印象中Openresty已默认支持）：ngx_http_substitutions_filter_module，可以通过正则表达式来替换内容：
+
+```Nginx
+subs_filter_types text/html text/css text/xml;
+subs_filter st(\d*).example.com $1.example.com ir;
+subs_filter a.example.com s.example.com;
+subs_filter http://$host https://$host;
+```
+
+但是 APISIX 的 response-rewrite 插件文档，好像只能支持完整替换，相当于直接将所有响应都换成插件设置的返回，并不支持部分内容替换:
```
Comment (yzeng25) [inline website/i18n/zh/docusaurus-plugin-content-blog/2021/12/02/weekly-report-1130.md]:
```suggestion
但是 APISIX 的 response-rewrite 插件文档，好像只能支持完整替换，相当于直接将所有响应都换成插件设置的返回，并不支持部分内容替换：
```

File: website/blog/2021/12/02/weekly-report-1130.md [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+
+ - **多语言支持**
+-  - Apache APISIX 是一个通过 `WASM` 和 `RPC` 支持不同语言来进行插件开发的网关.
++  - Apache APISIX 是一个通过 `RPC` 和 `WASM` 支持不同语言来进行插件开发的网关.
+   ![Multi Language Support into Apache APISIX](../../../docs/assets/images/apisix-multi-lang-support.png)
+-  - WASM 或 WebAssembly 是比较现代的开发方式。 APISIX 能加载运行使用[Proxy WASM SDK](https://github.com/proxy-wasm/spec#sdks)编译的 WASM 字节码。开发者仅需要使用该 SDK 编写代码，然后编译成 WASM 字节码，即可运行在 APISIX 中的 WASM 虚拟机中。
+-  - RPC 是一种比较传统的开发方式。开发者可以使用他们需要的语言来进行 RPC 服务的开发，该 RPC 通过本地通讯来跟 APISIX 进行数据交换。到目前为止，APISIX 已支持[Java](https://github.com/apache/apisix-java-plugin-runner), [Golang](https://github.com/apache/apisix-go-plugin-runner), [Python](https://github.com/apache/apisix-python-plugin-runner) and Node.js.
++  - RPC 是当前采用的开发方式。开发者可以使用他们需要的语言来进行 RPC 服务的开发，该 RPC 通过本地通讯来跟 APISIX 进行数据交换。到目前为止，APISIX 已支持[Java](https://github.com/apache/apisix-java-plugin-runner), [Golang](https://github.com/apache/apisix-go-plugin-runner), [Python](https://github.com/apache/apisix-python-plugin-runner) and Node.js.
++  - WASM 或 WebAssembly 是实验性的开发方式。 APISIX 能加载运行使用[Proxy WASM SDK](https://github.com/proxy-wasm/spec#sdks)编译的 WASM 字节码。开发者仅需要使用该 SDK 编写代码，然后编译成 WASM 字节码，即可运行在 APISIX 中的 WASM 虚拟机中。
```
Comment (yzeng25) [inline website/blog/2021/12/02/weekly-report-1130.md]: I am not sure if this part should be here...

File: website/i18n/zh/docusaurus-plugin-content-blog/2021/12/02/weekly-report-1130.md [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+   ![Multi Language Support into Apache APISIX](../../../docs/assets/images/apisix-multi-lang-support.png)
+-  - WASM 或 WebAssembly 是比较现代的开发方式。 APISIX 能加载运行使用[Proxy WASM SDK](https://github.com/proxy-wasm/spec#sdks)编译的 WASM 字节码。开发者仅需要使用该 SDK 编写代码，然后编译成 WASM 字节码，即可运行在 APISIX 中的 WASM 虚拟机中。
+-  - RPC 是一种比较传统的开发方式。开发者可以使用他们需要的语言来进行 RPC 服务的开发，该 RPC 通过本地通讯来跟 APISIX 进行数据交换。到目前为止，APISIX 已支持[Java](https://github.com/apache/apisix-java-plugin-runner), [Golang](https://github.com/apache/apisix-go-plugin-runner), [Python](https://github.com/apache/apisix-python-plugin-runner) and Node.js.
++  - RPC 是当前采用的开发方式。开发者可以使用他们需要的语言来进行 RPC 服务的开发，该 RPC 通过本地通讯来跟 APISIX 进行数据交换。到目前为止，APISIX 已支持[Java](https://github.com/apache/apisix-java-plugin-runner), [Golang](https://github.com/apache/apisix-go-plugin-runner), [Python](https://github.com/apache/apisix-python-plugin-runner) and Node.js.
++  - WASM 或 WebAssembly 是实验性的开发方式。 APISIX 能加载运行使用[Proxy WASM SDK](https://github.com/proxy-wasm/spec#sdks)编译的 WASM 字节码。开发者仅需要使用该 SDK 编写代码，然后编译成 WASM 字节码，即可运行在 APISIX 中的 WASM 虚拟机中。
+
+ - **Serverless**
+   - [Lua functions](plugins/serverless.md): 能在 APISIX 每个阶段调用 lua 函数.
+```
```
Comment (yzeng25) [inline website/i18n/zh/docusaurus-plugin-content-blog/2021/12/02/weekly-report-1130.md]: ditto, this part looks very strange, please double check

File: website/i18n/zh/docusaurus-plugin-content-blog/2021/12/02/weekly-report-1130.md [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+
+## 近期功能特性亮点
+
+- [kafka-logger 支持记录请求体](https://github.com/apache/apisix/pull/5501)（贡献者: [windyrjc](https://github.com/windyrjc)）
```
Comment (yzeng25) [inline website/i18n/zh/docusaurus-plugin-content-blog/2021/12/02/weekly-report-1130.md]:
```suggestion
- [kafka-logger 支持记录请求体](https://github.com/apache/apisix/pull/5501)（贡献者：[windyrjc](https://github.com/windyrjc)）
```

File: website/i18n/zh/docusaurus-plugin-content-blog/2021/12/02/weekly-report-1130.md [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+
+- [新增 azure-functions 插件，与 Azure Serverless Function 无缝集成](https://github.com/apache/apisix/pull/5479)（贡献者: [bisakhmondal](https://github.com/bisakhmondal)）
```
Comment (yzeng25) [inline website/i18n/zh/docusaurus-plugin-content-blog/2021/12/02/weekly-report-1130.md:97]:
```suggestion
- [新增 azure-functions 插件，与 Azure Serverless Function 无缝集成](https://github.com/apache/apisix/pull/5479)（贡献者：[bisakhmondal](https://github.com/bisakhmondal)）
```

File: website/i18n/zh/docusaurus-plugin-content-blog/2021/12/02/weekly-report-1130.md [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+
+- [WASM 插件支持在 header_filter 阶段运行](https://github.com/apache/apisix/pull/5544)（贡献者: [spacewander](https://github.com/spacewander)）
```
Comment (yzeng25) [inline website/i18n/zh/docusaurus-plugin-content-blog/2021/12/02/weekly-report-1130.md]: ditto ：

File: website/i18n/zh/docusaurus-plugin-content-blog/2021/12/02/weekly-report-1130.md [hunk 9]
```diff
@@ -0,0 +1,174 @@
...
+
+- [新增 google-cloud-logging 插件，用于推送日志到 Google Cloud logging Service](https://github.com/apache/apisix/pull/5538)（贡献者: [shuaijinchao](https://github.com/shuaijinchao)）
```
Comment (yzeng25) [inline website/i18n/zh/docusaurus-plugin-content-blog/2021/12/02/weekly-report-1130.md]: ditto ：

File: website/i18n/zh/docusaurus-plugin-content-blog/2021/12/02/weekly-report-1130.md [hunk 10]
```diff
@@ -0,0 +1,174 @@
...
+
+- [新增 openwhisk 插件，与 Apache OpenWhisk serverless 平台集成](https://github.com/apache/apisix/pull/5518)（贡献者: [bzp2010](https://github.com/bzp2010)）
```
Comment (yzeng25) [inline website/i18n/zh/docusaurus-plugin-content-blog/2021/12/02/weekly-report-1130.md]: ：

File: website/i18n/zh/docusaurus-plugin-content-blog/2021/12/02/weekly-report-1130.md [hunk 11]
```diff
@@ -0,0 +1,174 @@
...
+
+- [kafka-logger 和 http 支持记录响应体](https://github.com/apache/apisix/pull/5550)（贡献者: [dmsolr](https://github.com/dmsolr)）
```
Comment (yzeng25) [inline website/i18n/zh/docusaurus-plugin-content-blog/2021/12/02/weekly-report-1130.md]: ：

File: website/i18n/zh/docusaurus-plugin-content-blog/2021/12/02/weekly-report-1130.md [hunk 12]
```diff
@@ -0,0 +1,174 @@
...
+
+- [在 APISIX Ingress 中对于 HTTPS 和 gRPCs 类型的 upstream 丰富了 mTLS 的支持](https://github.com/apache/apisix-ingress-controller/pull/755)（贡献者: [nic-6443](https://github.com/nic-6443)）
```
Comment (yzeng25) [inline website/i18n/zh/docusaurus-plugin-content-blog/2021/12/02/weekly-report-1130.md:107]: ：

**Review by bzp2010 — APPROVED** [review]

**Review by liuxiran — APPROVED** [review]

**Pull request merged**